src/test/frametest.cpp

```diff
@@ -0,0 +1,31 @@
+#include <gtest/gtest.h>
+
+#include <QDebug>
+
+#include "audio/frame.h"
+
+class FrameTest : public testing::Test {
+};
+
+TEST_F(FrameTest, TestFramePosValid) {
+    EXPECT_TRUE(mixxx::audio::FramePos(100).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(2000).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(0).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(-1).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(99999).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(-99999).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(128.5).isValid());
+    EXPECT_TRUE(mixxx::audio::FramePos(135.67).isValid());
+    EXPECT_FALSE(mixxx::audio::FramePos().isValid());
+}
+
+TEST_F(FrameTest, TestFramePosFractional) {
+    EXPECT_FALSE(mixxx::audio::FramePos(100).isFractional());
+    EXPECT_FALSE(mixxx::audio::FramePos(2000).isFractional());
+    EXPECT_FALSE(mixxx::audio::FramePos(0).isFractional());
+    EXPECT_FALSE(mixxx::audio::FramePos(-1).isFractional());
+    EXPECT_FALSE(mixxx::audio::FramePos(99999).isFractional());
+    EXPECT_FALSE(mixxx::audio::FramePos(-99999).isFractional());
+    EXPECT_TRUE(mixxx::audio::FramePos(128.5).isFractional());
+    EXPECT_TRUE(mixxx::audio::FramePos(135.67).isFractional());
```

Comment:
Please add her also a check with Nan values and denormals.

Reply:
What should it return? Is nan fractional?

Reply:
Let's accept that the result is undefined. It doesn't matter and passing NaN is a bug. Just add a debug assertion to detect this case.

Reply:
done.